path/walker.go

```diff
@@ -16,29 +16,67 @@
 
 package path
 
-// Type Visitor is a function which performs an action on the provided path.
-// The return value is a slice of subdirectories to visit, a function to call to "close"
-// the visited directory on successful visitation or an error.
-type Visitor func(string) ([]string, func() error, error)
+// Visitor is a interface which performs an action on the provided string-valued path.
+type Visitor interface {
```

Comment:
Do we need both kinds ("country _and_ western")? It looks like the library does ~everything in terms of Path values anyway.

If we really do need both, I'd advocate for using the special name for the string-based one, since `path.PathVisitor` is kind of redundant whereas `path.StringVisitor` would avoid that. But the examples seem to only use the path one.

Reply:
Not really, no. Removed.